src/components/Timer/Timer.js

```diff
@@ -0,0 +1,75 @@
+import React, { useState, useEffect } from 'react';
+
+import styled from 'styled-components';
+
+const Timer = () => {    
+    const [minute, setMinute] = useState(0);
+    const [second, setSecond] = useState(0);
+    const [mill, setMill] = useState(0);
+    const [time, setTime] = useState("00:00:00")
+    const [stopwatchStart, setStopwatchStart] = useState(false);
+    const [stopwatchTime, setStopwatchTime] = useState(0);
+    
+    const handleKeyDown = event => {
+        if (event.keyCode === 32){
+
+            if (!stopwatchStart) {
+                setStopwatchTime(Date.now());
+                setStopwatchStart(true);
+            }
+
+            const timeGap = Date.now() - stopwatchTime;
+            setMill(Math.floor((timeGap % 1000) / 10));
+            setSecond(Math.floor(timeGap / 1000));
+            setMinute(Math.floor(timeGap / 60000));
+            setTime(() => {
+                let resultTime = "";
+                if (minute < 10) resultTime += '0';
+                resultTime += minute + ':';
+                if (second < 10) resultTime += '0';
+                resultTime += second + ':';
+                if (mill < 10) resultTime += '0';
+                resultTime += mill;
+                return resultTime;
+            });
+        }
+    }
+
+    const handleKeyUp = event => {
+        if (event.keyCode === 32) {
+            setTime("00:00:00");
+            setMill(0);
+            setSecond(0);
+            setMinute(0);
+            setStopwatchStart(false);
+            setStopwatchTime(0);
+        }
+    }
```

Comment:
Your default behavior now follows this patter:
- I press key down and it (should) start counting.
- But as soon as I let the key go it immediately clears
Is it expected behavior? I recommend starting with simple timer, and add countdown later.

Reply:
- `onKeyDown` means **while key is down** counting is happening.
- And when you let go it clears. 
It is expected behavior of the timer.
What once was planned as a timer now became a stopwatch with timer behavior...
I'll fix it.